content/en/docs/languages/java/sdk.md

```diff
@@ -0,0 +1,1386 @@
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    // Shutdown the exporter and cleanup any resources.
+    logger.log(Level.INFO, "shutting down");
+    return CompletableResultCode.ofSuccess();
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+#### SpanLimits
+
+[SpanLimits](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-trace/latest/io/opentelemetry/sdk/trace/SpanLimits.html)
+defines constraints for the data captured by spans, including max attribute
+length, max number of attributes, and more.
+
+The following code snippet demonstrates `SpanLimits` programmatic configuration:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/SpanLimitsConfig.java"?>
+```java
+package otel;
+
+import io.opentelemetry.sdk.trace.SpanLimits;
+
+public class SpanLimitsConfig {
+  public static SpanLimits spanLimits() {
+    return SpanLimits.builder()
+        .setMaxNumberOfAttributes(128)
+        .setMaxAttributeValueLength(1024)
+        .setMaxNumberOfLinks(128)
+        .setMaxNumberOfAttributesPerLink(128)
+        .setMaxNumberOfEvents(128)
+        .setMaxNumberOfAttributesPerEvent(128)
+        .build();
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+### SdkMeterProvider
+
+[SdkMeterProvider](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-metrics/latest/io/opentelemetry/sdk/metrics/SdkMeterProvider.html)
+is the SDK implementation of
+[MeterProvider](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/MeterProvider.html),
+and is responsible for handling metric telemetry produced by the API.
+
+`SdkMeterProvider` is configured by the application owner, and consists of:
+
+- [Resource](#resource) the resource metrics are associated with.
+- [MetricReader](#metricreader) to read the aggregated state of metrics.
+- [MetricExporter](#metricexporter) to export metrics out of process, and which
+  are associated with `MetricReader`s.
+- [Views](#views) to configure metric streams, including dropping unused
+  metrics.
+
+The following code snippet demonstrates `SdkMeterProvider` programmatic
+configuration:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/SdkMeterProviderConfig.java"?>
+```java
+package otel;
+
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.List;
+import java.util.Set;
+
+public class SdkMeterProviderConfig {
+  public static SdkMeterProvider create(Resource resource) {
+    SdkMeterProviderBuilder builder =
+        SdkMeterProvider.builder()
+            .setResource(resource)
+            .registerMetricReader(
+                MetricReaderConfig.periodicMetricReader(
+                    MetricExporterConfig.otlpHttpMetricExporter(
+                        "http://localhost:4318/v1/metrics")));
```

Comment:
@open-telemetry/docs-maintainers - these code snippets are great, really transforms how I think about writing docs knowing that I can write / maintain snippets which are well formatted and which I know will compile. 

One thing I find myself wanting is a link to source code where the snippet actually lives. In instances like this, it's nice for the user to be able to go look at the source code to see what `MetricExporterConfig.otlpHttpMetricExporter` does. I keep it as a separate function in a separate file to avoid repeating the content which lives down in the `#### Metric Exporter`. 

Is this something you've thought about? Seems like the code-excerpt tool would have access to the information necessary to add a link to source.

Reply:
If you can open an issue to capture that need, I can look into it some time this fall when I get back into code-excerpt tooling support.

Reply:
👍 Opened #5024 to track.